astropy/coordinates/distances.py

```diff
@@ -198,9 +198,10 @@ def _convert_to_and_validate_length_unit(unit, allow_dimensionless=False):
     """
     try:
         unit = u.Unit(unit)
-        assert (unit.is_equivalent(u.kpc) or
```

Comment:
This is a good example where I think `assert` is the correct usage: in normal code, the unit put in will be a `unit` of distance, and since it is expensive to check whether a unit is in fact a length, it is a good optimization to skip checking for that (and anything with the wrong input will fail badly somewhere else anyway).